### PR TITLE
Pass --local always when in GitHub Actions environment

### PR DESCRIPTION
**Pull request by dawidd6**
See discussion in: https://github.com/Homebrew/homebrew-core/pull/113506

Closes: https://github.com/Homebrew/homebrew-core/pull/113506

### CODE REVIEW

File: cmd/test-bot.rb [hunk 1]
```diff
@@ -106,14 +106,12 @@ def test_bot
...
     github_actions = ENV["GITHUB_ACTIONS"].present?
     if github_actions
-      ARGV << "--cleanup"
+      ARGV << "--cleanup" << "--local"
       ENV["HOMEBREW_COLOR"] = "1"
       ENV["HOMEBREW_GITHUB_ACTIONS"] = "1"
     end
 
     # TODO: remove me after a few days
     ENV["GITHUB_ACTIONS_HOMEBREW_SELF_HOSTED"] = "1" if ENV["GITHUB_ACTIONS_HOMEBREW_MACOS_SELF_HOSTED"].present?
```
Comment (MikeMcQuaid) [inline cmd/test-bot.rb:115]:
```suggestion
```

**Review by MikeMcQuaid — APPROVED** [review]
Fine with me! Be ready to rollback if we need to. Would be nice to remove the TODO now too.

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by dawidd6** [comment]
MacOS CI fails, can't figure out why.

**Comment by MikeMcQuaid** [comment]
@dawidd6 it's trying to write logs somewhere it doesn't have permission to: 
```
    Errno::EPERM: Operation not permitted @ rb_sysopen - /usr/local/Homebrew/Library/Taps/homebrew/homebrew-test-bot/logs/testbottest/post_install.01.testbottest
```

**Comment by MikeMcQuaid** [comment]
(it's also a bad idea to be writing logs to e.g. inside `/usr/local` anyway)

**Comment by MikeMcQuaid** [comment]
@dawidd6 We shouldn't be putting bottle files there. They should be in e.g. $HOME

**Comment by Bo98** [comment]
I'm a bit concerned with the compatibility of this change. Particularly since most taps will be using `brew test-bot` (we configure it in `brew tap-new`).

If this is requiring changes to workflows then I don't think it's worth it and could be passed in Homebrew/core like we do with many other things like `--junit`. Having to change the workflow here is already an indicator.

**Comment by Bo98** [comment]
Honestly the main reason `--local` is forced on this side is because of the non-ephemeral nature of our self-hosted runners, so that conditional can probably go when fully ephemeral.

Beyond that it's only useful because of the specific way we upload artifacts on Homebrew/core, so probably makes sense controlled there.

**Comment by dawidd6** [comment]
Yea I agree, we should fix this on the core tap level rather than here. Imma close this PR and go back to the previous one there.

**Comment by MikeMcQuaid** [comment]
@dawidd6 Could you reopen just to remove the TODO? Thanks 🙇🏻

**Comment by dawidd6** [comment]
https://github.com/Homebrew/homebrew-test-bot/pull/845

**Pull request closed without merging**